akka-actor-typed-tests/src/test/scala/akka/actor/typed/AskSpec.scala

```diff
@@ -119,17 +121,19 @@ class AskSpec extends ScalaTestWithActorTestKit("""
     }
 
     "fail the future if the actor has terminated" in {
-      val actor: ActorRef[Msg] = spawn(behavior)
+      implicit val timeout: Timeout = 100.millis
 
+      val actor: ActorRef[Msg] = spawn(behavior)
       val deadLetterProbe = createDeadLetterProbe()
-      val probe = createTestProbe[Any]()
-      actor ! Stop(probe.ref)
-      implicit val timeout: Timeout = 10.millis
+      val terminated: Future[Unit] = actor.ask(Stop(_))
+      Await.result(terminated, 100.millis)
+      // give actor some time to terminated.
+      Thread.sleep(100)
```

Comment:
fix not robustness on #32104

Reply:
Using the probe to block the test until termination has propagated would be more efficient than a 100ms sleep... in fact this test case can be merged with the case starting at line 60 which tests the same scenario (ask of an already-terminated actor).

Reply:
Great input of test, thanks!

Reply:
Solved on #32111